File: webcam.cpp
// webcam.cpp
#include <opencv2/opencv.hpp>
#include <iostream>
#include <unistd.h>

using namespace cv;
using namespace std;

int main(int argc, char** argv) {
    string output = "snapshot.jpg";
    int device = 0, delay = 0;

    for (int i = 1; i < argc; ++i) {
        string arg = argv[i];
        if (arg == "-o" && i+1 < argc) output = argv[++i];
        else if (arg == "-d" && i+1 < argc) device = stoi(argv[++i]);
        else if (arg == "-t" && i+1 < argc) delay = stoi(argv[++i]);
        else if (arg == "-h" || arg == "--help") {
            cout << "Usage: webcam [-o output] [-d device] [-t delay]" << endl;
            return 0;
        }
    }

    VideoCapture cap(device);
    if (!cap.isOpened()) {
        cerr << "Ошибка: не удалось открыть камеру" << endl;
        return 1;
    }

    if (delay > 0) {
        cout << "Ожидание " << delay << " секунд..." << endl;
        sleep(delay);
    }

    Mat frame;
    cap >> frame;
    if (frame.empty()) {
        cerr << "Ошибка: не удалось захватить кадр" << endl;
        return 1;
    }

    imwrite(output, frame);
    cout << "✅ Изображение сохранено в " << output << endl;
    return 0;
}

File: webcam.py
# webcam.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import cv2
import argparse
import time
import sys

def main():
    parser = argparse.ArgumentParser(description="Захват изображения с веб-камеры")
    parser.add_argument('-o', '--output', default='snapshot.jpg', help='Выходной файл')
    parser.add_argument('-d', '--device', type=int, default=0, help='Индекс камеры')
    parser.add_argument('-t', '--delay', type=int, default=0, help='Задержка в секундах')
    args = parser.parse_args()

    cap = cv2.VideoCapture(args.device)
    if not cap.isOpened():
        print("Ошибка: не удалось открыть камеру", file=sys.stderr)
        sys.exit(1)

    if args.delay > 0:
        print(f"Ожидание {args.delay} секунд...")
        time.sleep(args.delay)

    ret, frame = cap.read()
    cap.release()

    if not ret:
        print("Ошибка: не удалось захватить кадр", file=sys.stderr)
        sys.exit(1)

    cv2.imwrite(args.output, frame)
    print(f"✅ Изображение сохранено в {args.output}")

if __name__ == "__main__":
    try:
        main()
    except KeyboardInterrupt:
        print("\nПрервано пользователем.")
        sys.exit(0)
